client/platform/desktop/backend/native/common.ts

```diff
@@ -545,41 +546,53 @@ async function saveAttributes(settings: Settings, datasetId: string, args: SaveA
 }
 
 
-async function processAnnotationFilePath(
+async function _ingestFilePath(
   settings: Settings,
   datasetId: string,
   path: string,
-) {
+): Promise<(DatasetMetaMutable & { fps?: number }) | null> {
   if (!fs.existsSync(path)) {
-    return {};
+    return null;
+  }
+  if (fs.statSync(path).size === 0) {
+    return null;
   }
-  if (fs.statSync(path).size > 0) {
-    // Attempt to process the file
-    let data: viameSerializers.AnnotationFileData;
-    if (npath.extname(path) === '.json' && await nistSerializers.confirmNistFile(path)) {
-      data = await nistSerializers.loadNistFile(path);
+  // Make a copy of the file in aux
+  const projectInfo = getProjectDir(settings, datasetId);
+  const newPath = npath.join(projectInfo.auxDirAbsPath, `imported_${npath.basename(path)}`);
+  await fs.copy(path, newPath);
+  // Attempt to process the file
+  let tracks: TrackData[] | null = null;
+  const meta: DatasetMetaMutable & { fps?: number } = {};
+  if (JsonFileName.test(path)) {
+    const jsonObject = await _loadAsJson(path);
+    if (nistSerializers.confirmNistFormat(jsonObject)) {
+      // NIST json file
+      const data = await nistSerializers.loadNistFile(path);
+      tracks = data.tracks;
+      meta.fps = data.fps;
+    } else if (DatasetMetaMutableKeys.some((key) => key in jsonObject)) {
```

Comment:
We need to restrict to only the `DatasetMetaMutableKeys`.  If you do a full export from the server you get a `meta.json` file.  If you import that file it passes this check but then merges with the existing meta structure causing the file to have a bunch of extra fields which may be incorrect if there is any overlap (like `fps` and `type` fields).  This could also occur if a user tries to import their meta.json desktop file from another or similar dataset.  It will do the merge and completely break the data then.

Reply:
* Broken test case added in 06a7815
* Test fixed in 98d159f